src/extension/providers/debug_config_provider.ts

```diff
@@ -680,7 +680,9 @@ export class DebugConfigProvider implements DebugConfigurationProvider {
 					this.addArgsIfNotExist(args, "--web-renderer", renderer);
 			}
 
-			if (this.wsContext.config.forceFlutterWorkspace && conf.daemonPort) {
+			if (this.wsContext.config.forceFlutterWorkspace && portFromLocalExtension) {
+				this.addArgsIfNotExist(args, "--daemon-connection-port", portFromLocalExtension.toString());
+			} else if (this.wsContext.config.forceFlutterWorkspace && conf.daemonPort) {
```

Comment:
Maybe worth combining these two ports into a single variable to slightly reduce duplication of the arg?

```dart
const daemonPort = this.deviceManager?.portFromLocalExtension ?? conf.daemonPort;
if (this.wsContext.config.forceFlutterWorkspace && daemonPort) {
	this.addArgsIfNotExist(args, "--daemon-connection-port", daemonPort.toString());
}
```